packages/app/src/components/search/SearchModal.tsx

```diff
@@ -200,10 +186,7 @@ export const SearchModal = ({ toggleModal }: { toggleModal: () => void }) => {
             </Grid>
           </Grid>
           <Grid item xs>
-            <SearchResult
-              onClick={handleSearchResultClick}
-              onKeyDown={handleSearchResultClick}
-            >
+            <SearchResult onClick={toggleModal}>
```

Comment:
One thing that came to mind, would any click on the search result close the modal? Even if the click is not on the title link that redirects to the result page? (Appears to be a bug that already exists, unrelated to your changes)

Reply:
I can think of a couple of situations where the click happens somewhere else. For example bread crumbs links, or if one decide to have for example other kind of metadata e.g. owner. In those situations I think it should be closed as well. Were you thinking of any other situations where a click on the search result should not result in a modal close?

Reply:
Maybe before toggle we can check that the element clicked on is a link or button? what do you think?

Reply:
In a case where we have a collapsable result item layout, where people can expand the result item to see more details, for example, and it doesn't necessarily close the modal 🤔?
A suggestion can be that the result items receive a prop like `onResultClick` which can be a callback that closes the modal, or we always close the modal if the location changes.
It might be a separate issue, not blocking the improvements you've made here 🙂 .